### PR TITLE
add validateAddress

### LINKED ISSUE #466: allow address validation

### PR DESCRIPTION
**Pull request by Elli610**
closes #466

### CODE REVIEW

File: src/web3/WalletClient.ts [hunk 1]
```diff
@@ -564,4 +569,25 @@ export class WalletClient extends BaseClient implements IWalletClient {
...
+
+  /**
+   * Validate an address.
+   *
+   * @param address - The address to validate.
+   *
+   * @returns A boolean (true if the address is valid, false otherwise).
+   */
+  validateAddress(address: string): boolean {
+    address = address.slice(ADDRESS_PREFIX.length); // remove The address prefix
```
Comment (Ben-Rey) [inline src/web3/WalletClient.ts:581]: Should we check for the right address prefix ?

**Review by Ben-Rey — CHANGES_REQUESTED** [review]

File: src/web3/WalletClient.ts [hunk 2]
```diff
@@ -564,4 +569,25 @@ export class WalletClient extends BaseClient implements IWalletClient {
...
+    // check if the address is base58 encoded
+    try {
+      const bytesAddress = new Uint8Array(base58Decode(address));
+      if (bytesAddress.length != 32) {
```
Comment (Ben-Rey) [inline src/web3/WalletClient.ts:585]: In the rust side they check if the length of the address, `without the version` is equal to 32

And the version is an U64 varint

@sydhds please correct me if I misunderstood something

Comment (Elli610) [reply src/web3/WalletClient.ts]: updated !

File: src/web3/WalletClient.ts [hunk 3]
```diff
@@ -564,4 +570,28 @@ export class WalletClient extends BaseClient implements IWalletClient {
...
+   *
+   * @returns A boolean (true if the address is valid, false otherwise).
+   */
+  validateAddress(address: string): boolean {
+    address = address.slice(ADDRESS_PREFIX.length); // remove The address prefix
+    // check if the address is base58 encoded
+    try {
+      // remove the version byte
+      const versionAndKeyBytes = new Uint8Array(base58Decode(address));
+      varintDecode(versionAndKeyBytes.slice(0, 1)); // check the version byte
```
Comment (sydhds) [inline src/web3/WalletClient.ts:587]: it seems that the varintDecode is done only on 1 byte. In Rust code, the version is encoded as a u64 varint so I guess its size could be more than 1 byte no?

**Comment by Ben-Rey** [comment]
should be done here https://github.com/massalabs/massa-web3/pull/470

**Pull request closed without merging**